### ISSUE TITLE
Cost, Profit and receiving quantity.

### ISSUE THREAD

**Issue by tanakornp** [jekkos/opensourcepos#936]
### New Feature or Enhacement

ospos_sales_items.item_cost_price is not reflect the the actual cost when receiving quantity of item is not 1.

For example, I'm selling a rubber gloves.

The item came in as a dozen package so the receiving quantity is 12.
Cost of item for 1 dozen is 210 THB (Thai Baht) or 210/12 = 17.5 THB each.

I sell each glove at 30 THB.

Each sell of globe I got a profit of 30-17.5 = 12.5 THB
However, in ospos, it shows as I lost 30-210 = -180 THB

ospos_sales_items.item_cost_price was pull ospos_items.cost_price and didn't divide by ospos_items.receiving_quantity

**Comment by daN4cat** [comment]
I'm not 100% sure but I don't think receiving has a concept of bulk receiving and detailed sale.
This is something new to add probably.

**Comment by alvin101** [comment]
What about landed cost?  Your profit will not be correct without it.
I do not believe that profit or loss belong in POS, it should be left to back office or erp. 
Managing purchases orders, inventory, and accounting should all be back office, not POS,

Maybe it is time to consider a back office module for those that need more features than just a fast cash register.

**Comment by tanakornp** [comment]
Well there are a bunch of report in OSPOS does show the profit. I didn't want to do all the accounting here, I just only use for a estimated profit. I just want to see these features that already available here (receiving quantity and profit calculation) works well together.

**Comment by tanakornp** [comment]
@daN4cat 
Yeah, I suspect that too but from way of the receiving and receiving quantity work, I only way I see that can fit is bulk receiving. So I assumed you guys probably designed this for bulk receiving.

**Comment by jekkos** [comment]
Check #78 for a full explanation on what the feature does.

**Comment by jekkos** [comment]
The concept of profit here is only taking the difference between cost and unit price per item into account. AFAIK it does not take into account the amount paid on orders and potential extra discount you received on it. 

This is something that would be nice to have for the future. Currently it might be better to omit this field from the reporting as it isn't really a very exact representation of profit in most situations.

**Comment by tanakornp** [comment]
@jekkos 

Agreed . I have never cared about profit coz I knew that it is not accurate since 2.4. And I see no issue addressed this problem so post this issue. Just in case nobody knows it.

**Comment by jekkos** [comment]
You're right, I have ignored it ever since I took over the project as it never made sense to me.

Might be better to remove it unless someone thinks it is useful in some way?

**Comment by daN4cat** [comment]
I would leave the profit figure as it is, it's indicative and not the real operative figure you would put in a financial statement for sure.

**Comment by alvin101** [comment]
I vote to remove it.

**Comment by daN4cat** [comment]
We can make it optional :-)
